packages/frontend/src/components/ChartConfigPanel/FieldLayoutOptions.tsx

```diff
@@ -173,33 +189,80 @@ const FieldLayoutOptions: FC<Props> = ({ items }) => {
             >
                 <AxisGroup>
                     <AxisTitle>Group</AxisTitle>
-                    <AxisFieldDropdown>
-                        <FieldAutoComplete
-                            fields={availableDimensions}
-                            placeholder="Select a field to group by"
-                            activeField={
-                                chartHasMetricOrTableCalc
-                                    ? groupSelectedField
-                                    : undefined
+                    {pivotDimensions &&
+                        pivotDimensions.map((pivotKey) => {
+                            // Group series logic
+                            const groupSelectedField = availableDimensions.find(
+                                (item) => getItemId(item) === pivotKey,
+                            );
+
+                            return (
+                                <AxisFieldDropdown key={pivotKey}>
+                                    <FieldAutoComplete
+                                        fields={
+                                            groupSelectedField
+                                                ? [
+                                                      groupSelectedField,
+                                                      ...availableGroupByDimensions,
+                                                  ]
+                                                : availableGroupByDimensions
+                                        }
+                                        placeholder="Select a field to group by"
+                                        activeField={
+                                            chartHasMetricOrTableCalc
+                                                ? groupSelectedField
+                                                : undefined
+                                        }
+                                        onChange={(item) => {
+                                            setPivotDimensions(
+                                                pivotDimensions
+                                                    ? pivotDimensions.map(
```

Comment:
I don't understand what this is supposed to do...

Reply:
It is replacing the value in the array. I moved that logic to a util function so is readable 👍